crates/trigger/src/cli.rs

```diff
@@ -164,13 +165,28 @@ where
         let mut builder = TriggerExecutorBuilder::new(loader);
         self.update_wasmtime_config(builder.wasmtime_config_mut())?;
 
-        let logging_hooks =
-            StdioLoggingTriggerHooks::new(self.follow_components(), self.log.clone());
+        let logging_hooks = StdioLoggingTriggerHooks::new(self.follow_components(), self.log_dir());
         builder.hooks(logging_hooks);
 
         builder.build(locked_url, trigger_config).await
     }
 
+    pub fn log_dir(&self) -> Option<PathBuf> {
+        match &self.log {
+            Some(l) => return Some(l.clone()),
+            None => {
+                if self.from_registry {
+                    return None;
+                } else {
+                    let state_env = std::env::var(SPIN_STATE_DIR).context(SPIN_STATE_DIR).ok()?;
```

Comment:
The context here is not very human readable _grin_

Reply:
updated.